tests/testthat/test-reload.R

```diff
@@ -0,0 +1,155 @@
+        test_project <- tempfile('test_project')
+        suppressMessages(create.project(test_project, minimal = FALSE))
+        on.exit(unlink(test_project, recursive = TRUE), add = TRUE)
+        
+        oldwd <- setwd(test_project)
+        on.exit(setwd(oldwd), add = TRUE)
+        
+        clear(force=TRUE)
+        
+        # Create some R code and put in data directory 
+        CODE <- paste0(deparse(substitute({
+                x <- 10
+                
+        })), collapse ="\n")
+        
+        # save R code in the data directory
+        writeLines(CODE, "data/x.R")
+        
+        CODE <- paste0(deparse(substitute({
+                y <- 20
+                
+        })), collapse ="\n")
+        
+        # save R code in the data directory
+        writeLines(CODE, "data/y.R")
+        
+        CODE <- paste0(deparse(substitute({
+                z <- 10
+                
+        })), collapse ="\n")
+        
+        # save R code in the data directory
+        writeLines(CODE, "data/z.R")
+        
+        # Read the config data and set sticky_variables to x,y
+        config <- translate.dcf("config/global.dcf")
+        expect_error(config$sticky_variables <- "x,y", NA)
+        write.dcf(config, "config/global.dcf" )
+        
+        # Load the project and check that x exists and is in the cache
+        suppressMessages(load.project())
+        expect_true(exists("x"))
+        expect_true(.read.cache.info("x")$in.cache)
+        
+        # reload the project and config, x, y should not be cleared
+        expect_message(reload.project(), "not cleared: config x y")
+        
+        # Also, z should be reloaded from data
+        expect_true(exists("z"))
+        
+        
+})
+
+test_that('reload.project with reset clears everything', {
+        test_project <- tempfile('test_project')
+        suppressMessages(create.project(test_project, minimal = FALSE))
+        on.exit(unlink(test_project, recursive = TRUE), add = TRUE)
+        
+        oldwd <- setwd(test_project)
+        on.exit(setwd(oldwd), add = TRUE)
+        
+        clear(force=TRUE)
+        
+        # Create some R code and put in data directory 
+        CODE <- paste0(deparse(substitute({
+                x <- 10
+                
+        })), collapse ="\n")
+        
+        # save R code in the data directory
+        writeLines(CODE, "data/x.R")
+        
+        CODE <- paste0(deparse(substitute({
+                y <- 20
+                
+        })), collapse ="\n")
+        
+        # save R code in the data directory
+        writeLines(CODE, "data/y.R")
+        
+        CODE <- paste0(deparse(substitute({
+                z <- 10
+                
+        })), collapse ="\n")
+        
+        # save R code in the data directory
+        writeLines(CODE, "data/z.R")
+        
+        # Read the config data and set sticky_variables to x,y
+        config <- translate.dcf("config/global.dcf")
```

Comment:
I agree on testing user functions, but you're not testing the function (actually, that function isn't tested at all). I agree a wide spectrum of tests should be applied, but there is a distinction between the functions you use to setup the test environment (which should be tested separately and to which `translate.dcf` belongs in this case) and the functions you test in the test case. (Also, if you'd really want to simulate a texteditor why not use `readLines` and `writeLines`? ;-) )